packages/celotool/src/lib/blockscout.ts

```diff
@@ -129,16 +129,26 @@ async function helmParameters(
       )}`
     )
   }
-  if (isVmBased()) {
+  if (
+    fetchEnvOrFallback(envVar.BLOCKSCOUT_OVERRIDE_RPC_ENDPOINT, '') !== '' &&
+    fetchEnvOrFallback(envVar.BLOCKSCOUT_OVERRIDE_WS_ENDPOINT, '') !== ''
+  ) {
+    params.push(
+      `--set blockscout.jsonrpc_http_url=${fetchEnv(envVar.BLOCKSCOUT_OVERRIDE_RPC_ENDPOINT)}`
+    )
+    params.push(
+      `--set blockscout.jsonrpc_ws_url=${fetchEnv(envVar.BLOCKSCOUT_OVERRIDE_WS_ENDPOINT)}`
+    )
+  } else if (isVmBased()) {
     const txNodeLbIp = await getInternalTxNodeLoadBalancerIP(celoEnv)
     params.push(`--set blockscout.jsonrpc_http_url=http://${txNodeLbIp}:8545`)
     params.push(`--set blockscout.jsonrpc_ws_url=ws://${txNodeLbIp}:8546`)
   } else if (privateNodes > 0) {
     params.push(`--set blockscout.jsonrpc_http_url=http://tx-nodes-private:8545`)
     params.push(`--set blockscout.jsonrpc_ws_url=ws://tx-nodes-private:8546`)
   } else {
-    params.push(`--set blockscout.jsonrpc_http_url=http://tx-nodes-headless:8545`)
-    params.push(`--set blockscout.jsonrpc_ws_url=ws://tx-nodes-headless:8546`)
+    params.push(`--set blockscout.jsonrpc_http_url=http://tx-nodes:8545`)
```

Comment:
I'm not seeing where we went from `tx-nodes-headless` to `tx-nodes`. (Though I'm also not familiar with why/how this lookup works).

Reply:
For headless service, `kube-dns` returns a list of the pod's ips that are endpoints for that service, and the service does not have a service ip. In the regular service, the service has a cluster ip, kube-dns returns that ip for resolving the service, and `kube-proxy` routes the cluster-ip to a pod endpoint using iptables.
We have been switching between `headless` and regular services trying to improve the session affinity, and I think we have conclude is better to use regular services configured with sessionAffinity.